test/replication-luatest/gh_9943_promote_split_brain_test.lua

```diff
@@ -0,0 +1,124 @@
+local t = require('luatest')
+local replica_set = require('luatest.replica_set')
+local server = require('luatest.server')
+
+local g = t.group()
+
+g.before_all(function(g)
+    g.rs = replica_set:new({})
+    local rs_id = g.rs.id
+    g.box_cfg = {
+        replication_timeout = 0.1,
+        replication_synchro_timeout = 86400,
+        election_mode = 'manual',
+        replication = {
+            server.build_listen_uri('server1', rs_id),
+            server.build_listen_uri('server2', rs_id),
+            server.build_listen_uri('server3', rs_id),
+        },
+    }
+
+    for i = 1, 3 do
+        g.rs:build_and_add_server{
+            alias = 'server' .. i,
+            box_cfg = g.box_cfg,
+        }
+    end
+
+    g.rs:start()
+    g.rs:wait_for_fullmesh()
+    g.rs.servers[1]:exec(function() box.ctl.promote() end)
+    g.rs.servers[1]:wait_for_election_leader()
+    g.rs.servers[1]:exec(function()
+        box.schema.create_space('test', {is_sync = true}):create_index('pk')
+    end)
+    g.rs.servers[2]:wait_for_vclock_of(g.rs.servers[1])
+    g.rs.servers[3]:wait_for_vclock_of(g.rs.servers[1])
+end)
+
+g.after_all(function(g)
+    g.rs:stop()
+end)
+
+local function set_replication(instance, new_one)
+    instance:exec(function(repl)
+        box.cfg{replication = repl}
+    end, {new_one})
+end
+
+local function promote(instance)
+    instance:exec(function() box.ctl.promote() end)
+    instance:wait_for_election_leader()
+end
+
+local function test_promote_split_brain(g, with_data)
+    local new_replication = {
+        g.rs.servers[1].net_box_uri,
+        g.rs.servers[2].net_box_uri,
+    }
+
+    set_replication(g.rs.servers[3], {})
+    set_replication(g.rs.servers[1], new_replication)
+    set_replication(g.rs.servers[2], new_replication)
+
+    g.rs.servers[3]:exec(function()
+        box.cfg{replication_synchro_quorum = 1}
+    end)
+
+    promote(g.rs.servers[2])
+    g.rs.servers[1]:exec(function(id)
+        -- Without this wait PROMOTE may be not executed and exit with TimedOut
+        -- error. Let's make sure, that server_1 known about new leader.
+        t.helpers.retrying({}, function()
+            t.assert_equals(box.info.election.leader, id)
+        end)
+    end, {g.rs.servers[2]:get_instance_id()})
+    promote(g.rs.servers[1])
```

Comment:
Why did you need to promote 2 and then 1? Could you simply promote 1 right away? If this is to bump the term, then lets add a comment about that.

Reply:
> If this is to bump the term, then lets add a comment about that

it's, added comments